source/blog/2022-02-10-announcing-hanami-200alpha6.html.markdown

```diff
@@ -0,0 +1,166 @@
+---
+title: Announcing Hanami v2.0.0.alpha6
+date: 2022-02-10 11:04:32 UTC
+tags: announcements
+author: Luca Guidi
+image: true
+excerpt: >
+  Hanami Container Providers, Partial Slice Import/Export. Ruby 3.0+ only.
+---
+
+Hello Hanami community! We're thrilled to announce the release of Hanami 2.0.0.alpha6!
+
+With this new cycle of monthly based releases we have smaller set of changes, but delivered more frequently.
+
+## Hanami Container enhancements
+
+This month we focused mainly on the internals of the framework.
+The work that Tim Riley is doing is epic.
+Hanami 2 is modeled around `dry-system`, which powers the booting process and the dependencies of an app.
+
+### Providers
+
+First thing first, **we renamed _bootable compontents_ into _providers_**.
+This change was reflected into the public API.
+
+We also changed provider `init` into `prepare`.
+
+```ruby
+Hanami.application.register_provider(:metrics) do
+  prepare do
+    require "datadog/statsd"
+
+    # see application's config/settings.rb
+    @settings = settings.statsd
+  end
+
+  start do
+    register "metrics", Datadog::Statsd.new(@settings.host, @settings.port)
+  end
+
+  stop do
+    # triggered when the application is shut down
+    metrics.close
+  end
+end
```

Comment:
```suggestion
Hanami.application.register_provider(:metrics) do
  prepare do
    require "datadog/statsd"
  end

  start do
    # See application's config/settings.rb
    settings = target[:settings].statsd

    register "metrics", Datadog::Statsd.new(settings.host, settings.port)
  end

  stop do
    # triggered when the application is shut down
    metrics.close
  end
end
```

Turned it into something that'd actually work in theory :) And dropped the ivar as a way to share state between the steps since I think that's more of an advanced usage and felt more likely to confuse readers here than otherwise.